For upstream: -rdynamic is implemented by at least gcc and clang.
Using -export-dynamic with gcc only worked because of what was initially
a gcc implementation quirk.  -rdynamic isn't used at compile time anyway,
so using -rdynamic or -Wl,--export-dynamic in LDFLAGS would be enough.

Index: src/Makefile.am
--- src/Makefile.am.orig
+++ src/Makefile.am
@@ -32,7 +32,6 @@ klavaro_LDADD = \
 		-lgmodule-2.0
 
 if IS_POSIX
-AM_CPPFLAGS += -export-dynamic
 AM_LDFLAGS = -Wl,--export-dynamic
 endif
 
